### PR TITLE
Don't call result.ToString() in Random.AssertInRange

### LINKED ISSUE #52894: Random.Next is slow on checked runtime on ARM
This is a follow-up to the investigation in https://github.com/dotnet/runtime/issues/52031

We had a test that consistently run overtime on Linux on ARM whereas it was ok on all other platforms and architectures. Investigation shown that the culprit was calling for Random.Next in a loop (on checked runtime). Switching to Random.NextBytes changed timings dramatically.

There are two pieces of code with the timings I've measured on Ubuntu 18.04 on an ARM64 machine on Checked CoreCLR runtime:

```c#
int frameSize = 2 << 15;
byte[] message = new byte[frameSize * 10];
Random random = new(0);
for (int i = 0; i < message.Length; ++i)
{
    message[i] = (byte)random.Next(maxValue: 10);
}
```
Execution time: 4.8s

```c#
int frameSize = 2 << 15;
byte[] message = new byte[frameSize * 10];
new Random(0).NextBytes(message);
for (int i = 0; i < message.Length; ++i)
{
    message[i] %= 10;
}
```
Execution time: 0.02s

It might be beneficial to know what causes the first piece of code to run 200 times slower.

cc @danmoseley 

P.S.: it may be worth mentioning that on release runtime, the first piece takes 0.03s.

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes https://github.com/dotnet/runtime/issues/52894

Double.ToString() is quite slow in Checked (partially because of get_CurrentCulture calls + in general I guess Grisu is full of asserts)

### CODE REVIEW

**Comment by zlatanov** [comment]
Would it make sense to update the other assert method to do the same?

https://github.com/dotnet/runtime/blob/c5caf17083bef58c9575677a1c02a78db06d5e36/src/libraries/System.Private.CoreLib/src/System/Random.cs#L199-L210

**Comment by EgorBo** [comment]
> Would it make sense to update the other assert method to do the same?
> 
> https://github.com/dotnet/runtime/blob/c5caf17083bef58c9575677a1c02a78db06d5e36/src/libraries/System.Private.CoreLib/src/System/Random.cs#L199-L210

long to string is way faster but it's probably still make sense yeah

**Comment by SingleAccretion** [comment]
Heh, I suppose this would be a case where a custom string interpolation builder as the argument would help.

**Review by danmoseley — APPROVED** [review]
LGTM - thanks!

File: src/libraries/System.Private.CoreLib/src/System/Random.cs [hunk 1]
```diff
@@ -201,17 +201,29 @@ private static void AssertInRange(long result, long minInclusive, long maxExclus
         {
             if (maxExclusive > minInclusive)
             {
-                Debug.Assert(result >= minInclusive && result < maxExclusive, $"Expected {minInclusive} <= {result} < {maxExclusive}");
+                if (result < minInclusive || result >= maxExclusive)
+                {
+                    Debug.Assert(false, $"Expected {minInclusive} <= {result} < {maxExclusive}");
```
Comment (jkotas) [inline src/libraries/System.Private.CoreLib/src/System/Random.cs:206]:
```suggestion
                    Debug.Fail($"Expected {minInclusive} <= {result} < {maxExclusive}");
```
same for others

**Pull request merged**